user_manager/api/v1/views.py

```diff
@@ -0,0 +1,419 @@
+    """
+        **Use Case**
+
+            * Get a list of the managers that a user directly reports to.
+
+            * Add a manger for a user.
+
+            * Remove all managers for a user, or remove a single manager for a user.
+
+        **Example Request**
+
+            GET /api/user_manager/v1/managers/{user_id}/
+
+            POST /api/user_manager/v1/managers/{user_id}/ {
+                "email": "{email}"
+            }
+
+            DELETE /api/user_manager/v1/managers/{user_id}/
+
+            DELETE /api/user_manager/v1/managers/{user_id}/?user={user_id}
+
+        **GET Parameters**
+
+            * user_id: username or email address for user whose managers you want fetch
+
+        **POST Parameters**
+
+            * user_id: username or email address for user for whom you want to add a manger
+
+            * email: Email address for the manager
+
+        **DELETE Parameters**
+
+            * user_id: username or email address for manager
+
+        **GET Response Values**
+
+            If the request for information about the managers is successful, an HTTP 200 "OK"
+            response is returned with a collection of managers.
+
+            The HTTP 200 response has the following values.
+
+            * count: The number of managers in a course.
+
+            * next: The URI to the next page of results.
+
+            * previous: The URI to the previous page of results.
+
+            * num_pages: The number of pages.
+
+            * results: a list of managers directly over a user:
+
+                * id: The user id for a manager (may be null if manager hasn't
+                    registered an account).
+
+                * email: Email address of manager.
+
+        **Example GET Response**
+
+            GET /api/user_manager/v1/managers/staff@example.com/
+
+            {
+                "count": 99,
+                "next": "https://courses.org/api/user_manager/v1/managers/staff@example.com/?page=2",
+                "previous": null,
+                "results": {
+                    {
+                        "email": "edx@example.com",
+                        "id": 6
+                    },
+                    { ... }
+                }
+            }
+
+        **Example POST Response**
+
+            POST /api/user_manager/v1/managers/edx@example.com/ {
+                "email": "user@email.com"
+            }
+
+            {
+                "email": "user@email.com"
+                "id": 11
+            }
+
+        **Example DELETE Response**
+
+            DELETE /api/user_manager/v1/managers/edx@exmaple.com/
+
+            DELETE /api/user_manager/v1/managers/edx@exmaple.com/?report=some@user.com
+    """
+
+    serializer_class = UserManagerSerializer
+
+    @staticmethod
+    def _get_user_by_username_or_email(userid):
+        """
+        Get user by userid, which could be an email or username.
+        """
+
+        if '@' in userid:
```

Comment:
Did you check if `@` is excluded from usernames?

Reply:
Do you mean is something like 'h@t' a valid username? I wouldn't think so. These kinds of checks for the '@' symbol are littered throughout edx-platform.